static/js/accountSpecialist.js

```diff
@@ -197,3 +204,214 @@ function closeTagHandler(){
+    // function which handles adding services to list
+
+    var servicesEl = $('#service-selector');
+    var servicesList = servicesEl.find('#services_list');
+
+    function addService(title){
+        servicesList
+            .append(
+                '<div class="service-div">' +
+                    '<a class="collection-item blue-grey-text text-darken-1 ' +
+                        'service-item">' + title + '</a>' +
+                '</div>'
+        );
+    }
+
+    servicesList.empty();
+
+    //Add all services of selected category to list
+    _.forEach(services, function(item){
+        addService(item.title)
+    });
+
+    // Add services which were found by regex to list
+    $('#find_service').on('keyup', function(){
+        servicesList.empty();
+        var matches = findMatches($(this).val(),
+            _.map(services, 'title'));
+        if (matches.length > 0){
+            _.forEach(matches, function(service){
+                addService(service)
+            });
+        } else {
+            servicesList
+                .append('<a class="collection-item blue-grey-text' +
+                        ' text-darken-1">Cannot find service</a>')
+        }
+    });
+}
+
+function findMatches(q, strs){
+    // Regex for first letters match
+
+    var regexStartWith = new RegExp("^" + q, 'i');
+
+    var matches = [];
+    _.forEach(strs, function(str){
+        if (regexStartWith.test(str)) {
+            matches.push(str);
+        }
+    });
+
+    return matches
+}
+
+
+function categorySelectorHandler(categories, services){
+    // function which handles category selector
+
+    var card = $('#categories_card');
+    var body = $('body');
+
+    $('#show_category_dropdown').on('click', function(){
+        card.show()
+    });
+
+    // hide service selector when user click on body
+    $(document).mouseup(function (e) {
+        var container = $("#category-selector");
+
+        if (!container.is(e.target)
+            && container.has(e.target).length === 0) {
+            card.hide();
+        }
+    });
+
+    addCategoriesToList(categories);
+
+    // add category name to main input
+    body.on('click', '.category-item', function() {
+        $('#show_category_dropdown')
+            .val($(this).text())
+            .addClass('valid');
+        $('#show_service_dropdown')
+            .val('')
+            .prop('disabled', false)
+            .removeClass('valid');
+
+        $('#selected_service_id').val('');
+
+        card.hide();
+        var categoryId = parseInt(
+            $(this)
+                .closest('.category-div')
+                .find('input[class="category-id"]')
+                .val()
+        );
+        // call service selector handler with services for selected category
+        serviceSelectorHandler(_.filter(services, {'category_id': categoryId}))
+    })
+}
+
+
+function addCategoriesToList(categories){
+    // function which handles adding categories to list
```

Comment:
The comment here mostly repeats function name. It's better to describe why this function exists, for example